### PR TITLE
Replace print statements with logger

### PR DESCRIPTION
**Pull request by ramitsurana**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x ] I have read the **CONTRIBUTING.md** document.
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: zenml/repo/global_config.py [hunk 1]
```diff
@@ -106,10 +107,14 @@ def load(self):
         try:
             tbu = yaml_utils.read_json(self.path)
         except Exception as e:
-            print(str(e))
+            logger.error("Unable to load YAML file due to error: " + str(e) + ".Updating it as empty value.")
```
Comment (hamzamaiot) [inline zenml/repo/global_config.py:110]: > 80 characters. We follow [PEP-8 guidelines](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) for ZenML, so please add a line break before the first place e.g.

**Review by hamzamaiot — CHANGES_REQUESTED** [review]
Thanks @ramitsurana  ! Those prints spilled through, good to replace with proper logging. I just had one stylistic comment, so please do make that one change and ill merge right away.

Hope you are enjoying ZenML and hop on over to our Slack channel so we can say thanks directly: https://zenml.io/slack-invite !

File: zenml/repo/global_config.py (hunk 1 above)
Comment (ramitsurana) [reply zenml/repo/global_config.py]: Done

**Comment by hamzamaiot** [comment]
@ramitsurana  It seems like the tests are failing -> It feels like there is a circular import in the new imports you introduced. Can you take a look? Otherwise I can take a look a bit later in the day

File: zenml/utils/requirement_utils.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import re
-
 import subprocess
 import sys
+from zenml.logger import get_logger
```
Comment (htahir1) [inline zenml/utils/requirement_utils.py:4]: I think the circular dependency lives here @ramitsurana . If you import zenml here than it wont be able to install and might result in an error like https://github.com/maiot-io/zenml/runs/2437806331?check_suite_focus=true

Comment (htahir1) [reply zenml/utils/requirement_utils.py:4]: Rather do this import at the line 119

**Review by htahir1 — CHANGES_REQUESTED** [review]
Tests are still failing - left a comment for improving

**Comment by ramitsurana** [comment]
@htahir1  Cool. Thanks for letting me know. Apologies for the issue.

**Comment by htahir1** [comment]
All good - thanks for the PR!

**Pull request merged**